### PR TITLE
OCD-956: added role 'ROLE_ONC_STAFF'

### PR DESCRIPTION
**Pull request by dlucas1**
Started on this task and would like to get some feedback from Andrew. This is probably not the extent of changes necessary to complete the task. I also need to do some more testing.

### CODE REVIEW

File: app/admin/components/userEditCtrl.js [hunk 1]
```diff
@@ -26,14 +26,17 @@
                 if (!vm.acbId && !vm.atlId) {
                     vm.roles.push('ROLE_ADMIN');
                     vm.roles.push('ROLE_CMS_STAFF');
+                    vm.roles.push('ROLE_ONC_STAFF');
```
Comment (andlar) [inline app/admin/components/userEditCtrl.js]: This section of the file adds the ROLEs that a logged in user is able to give to someone else. In this case, Line 34 and 39 give ACB Admins and ATL Admins (respectively) the ability to give a user the ONC_STAFF role. That is not correct. Line 29 is the only place where this role should be added

File: app/admin/components/userManagementDirective.js [hunk 2]
```diff
@@ -17,14 +17,17 @@
                 if (!vm.acbId && !vm.atlId) { // not managing acb or atl, managing entire CHPL
                     vm.roles.push('ROLE_ADMIN');
                     vm.roles.push('ROLE_CMS_STAFF');
+                    vm.roles.push('ROLE_ONC_STAFF');
```
Comment (andlar) [inline app/admin/components/userManagementDirective.js]: Similarly to above, lines 25 & 30 should not add the ONC_STAFF role to the array of available ROLEs the logged in user can assign to a user

**Comment by dlucas1** [comment]
Made updates per Andrew's code review comments.

**Comment by andlar** [comment]
Still need RELEASE_NOTES updates, as well as changes to the HTML as to where the new `isOncStaff` function will be used

**Comment by dlucas1** [comment]
Still need to get the LHS navigation to show the CMS Management link. Everything else seems to be working as expected.

File: app/admin/admin.html [hunk 3]
```diff
@@ -49,7 +49,7 @@
 
             <li ng-class="{'active': vm.navState.screen === 'apiKeyManagement'}" ng-if="vm.isChplAdmin()"><a href="" class="" ng-click="vm.changeScreen('apiKeyManagement')">API Key Management</a></li>
             <li ng-class="{'active': vm.navState.screen === 'announcementsManagement'}" ng-if="vm.isChplAdmin()"><a href="" class="" ng-click="vm.changeScreen('announcementsManagement')">Announcements Management</a></li>
-            <li ng-class="{'active': vm.navState.screen === 'cmsManagement'}" ng-if="vm.isChplAdmin() || vm.isCmsStaff()"><a href="" class="" ng-click="vm.changeScreen('cmsManagement')">CMS Management</a></li>
+            <li ng-class="{'active': vm.navState.screen === 'cmsManagement'}" ng-if="vm.isChplAdmin() || vm.isCmsStaff() || vm.isOncStaff()"><a href="" class="" ng-click="vm.changeScreen('cmsManagement')">CMS Management</a></li>
```
Comment (andlar) [inline app/admin/admin.html]: This looks right, but the four Activity screens in the Reports section that are restricted to isChplAdmin() need to have an `|| vm.isOncStaff()` added in

File: app/admin/components/userEdit.html [hunk 4]
```diff
@@ -52,7 +52,7 @@ <h4 class="modal-title" ng-if="vm.action === 'invite'" id="editLabel">Invite Use
       </div>
       <div class="form-group">
         <label for="roles">User Role(s) <span class="text-danger">*</span></label>
-        <select ng-options="role as role for role in vm.roles" ng-model="vm.user.roles" multiple required size="5" id="roles" name="roles"></select>
+        <select ng-options="role as role for role in vm.roles" ng-model="vm.user.roles" multiple required size="6" id="roles" name="roles"></select>
```
Comment (andlar) [inline app/admin/components/userEdit.html]: I hadn't expanded the box when I added the CMS_STAFF role, so if you want to expand it to show all the roles at once you'll probably need to go to 7, not 6

File: app/admin/components/userManagement.html [hunk 5]
```diff
@@ -33,7 +33,7 @@
             <span><label for="isLocked_{{ user.user.userId }}">Locked:</label> <input type="checkbox" ng-model="user.user.accountLocked" disabled id="isLocked_{{ user.user.userId }}" /></span><br />
             <span><label for="isEnabled_{{ user.user.userId }}">Enabled:</label> <input type="checkbox" ng-model="user.user.accountEnabled" disabled id="isEnabled_{{ user.user.userId }}" /></span><br />
             <label for="roles_{{ user.user.userId }}">User Role<span ng-if="user.roles.length > 1">s</span>:</label>
-            <select ng-options="role as role for role in vm.roles" ng-model="user.roles" size="5" multiple disabled id="roles_{{ user.user.userId }}"></select>
+            <select ng-options="role as role for role in vm.roles" ng-model="user.roles" size="6" multiple disabled id="roles_{{ user.user.userId }}"></select>
```
Comment (andlar) [inline app/admin/components/userManagement.html]: Same as above. Likely want 7, not 6.

File: app/admin/admin.html [hunk 6]
```diff
@@ -41,10 +41,10 @@
...
-            <li class="subnav animate-if" ng-class="{'active': vm.navState.reports === 'users'}" ng-if="vm.navState.screen === 'reports' && vm.isChplAdmin()"><a href="" class="" ng-click="vm.changeSubNav('users')">Users</a></li>
-            <li class="subnav animate-if" ng-class="{'active': vm.navState.reports === 'api_key'}" ng-if="vm.navState.screen === 'reports' && vm.isChplAdmin()"><a href="" class="" ng-click="vm.changeSubNav('api_key_management')">API Key Management Activity</a></li>
-            <li class="subnav animate-if" ng-class="{'active': vm.navState.reports === 'api_key_usage'}" ng-if="vm.navState.screen === 'reports' && vm.isChplAdmin()"><a href="" class="" ng-click="vm.changeSubNav('api_key_usage')">API Key Usage Activity</a></li>
+            <li class="subnav animate-if" ng-class="{'active': vm.navState.reports === 'announcement'}" ng-if="vm.navState.screen === 'reports' && vm.isChplAdmin() || vm.isOncStaff()"><a href="" class="" ng-click="vm.changeSubNav('announcement')">Announcements</a></li>
```
Comment (andlar) [inline app/admin/admin.html]: Does this work? Or does it show this subnav if someone is ONCStaff but not on the Reports screen? As in, if they open the CMS Management page, will these four still show up? I think you might need to put the `vm.isChplAdmin() || vm.isOncStaff()` inside a set of parentheses

**Comment by dlucas1** [comment]
Please review and merge if all looks well

**Pull request merged**